src/withResolution/mediaQuery.spec.js

```diff
@@ -0,0 +1,79 @@
+import { register, unregister } from './mediaQuery'
+
+describe('Media Query Manager', () => {
```

Comment:
would we be able to do `spyOn(global.matchMedia)` (and/or `addListener`)  to validate that we are only adding as many listeners as we actually need?

Reply:
oh interesting, will do

Reply:
thinking about it more, we cannot attach a listener directly because `matchMedia` creates a new object to which you can attach queries. That's the part that's already tested